### PR TITLE
[LC] 26, 27, 58, 66, 70, 111, 118

### PR DESCRIPTION
**Pull request by azalexzhang**
### 📋 Summary

- Added Python solutions to LC 26, 27, 58, 66, 70 and 118
- Added C++ solutions to LC 111 and 118

### 🧠 Remember to

-   [ ] include tests
-   [x] add labels to pr
-   [x] tag reviewers
-   [ ] **after** squash merge, [reset your branch](https://github.com/LeetCube/GitCube/blob/main/pr/vsc.md#:~:text=fix%20commits%20ahead%20of%20main)

<details>
<summary>Draft PR</summary>

-   After initial code review, can convert to **Draft PR** to show your work is in progress
-   This disables merge, and reviewers won't receive notifications
-   After changes, click **Ready for Review**, and reviewers will receive notifications again

</details>
<details>
<summary>Reviewers</summary>

-   🟥 Request Changes: any must-do changes before merging
-   🟨 Comments: any optional changes before merging
-   🟩 Approve: if you are the last reviewer, merge it

</details>

### CODE REVIEW

File: data/contributors.json [hunk 1]
```diff
@@ -0,0 +1,116 @@
+[
```
Comment (dataprob) [inline data/contributors.json:1]: remove this file (which i don't see on my end hmmm)

File: doc/contributors/figs/langs.png
Comment (dataprob) [inline doc/contributors/figs/langs.png:1]: `doc/contributors/` is an old folder, remove the entire folder (i don't see this anymore on my end)

**Review by dataprob — COMMENTED** [review]
Great work alex! I'll take some time to review the solus. See my comments on file removal.

File: python/src/code_27.py
Comment (dataprob) [inline python/src/code_27.py:1]: Q desc says do it in-place, so try that way.

File: python/src/code_58.py [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import re
+
+
+class Solution:
+    def lengthOfLastWord(self, s: str) -> int:
+        strippedStr = s.strip()
```
Comment (dataprob) [inline python/src/code_58.py:6]: `s.strip()` strips all spaces and return a list of string. Then it becomes extremely simple!
`s = "   fly me   to   the moon  "`, `strippedStr == ['fly', 'me', 'to', 'the', 'moon']`

File: python/src/code_66.py [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+class Solution:
+    def plusOne(self, digits: List[int]) -> List[int]:
+        i = len(digits) - 1
+
+        while i >= 0:
+            digits[i] += 1
+
+            if digits[i] < 10:
+                break
```
Comment (dataprob) [inline python/src/code_66.py:12]: `return digits`

File: python/src/code_66.py [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+
+            digits[i] = 0
+            i -= 1
+
+        if i == -1:
```
Comment (dataprob) [inline python/src/code_66.py:17]: remove the `if` check

File: python/src/code_66.py [hunk 5]
```diff
@@ -0,0 +1,20 @@
+from typing import List
+
+
+class Solution:
+    def plusOne(self, digits: List[int]) -> List[int]:
+        i = len(digits) - 1
+
+        while i >= 0:
```
Comment (dataprob) [inline python/src/code_66.py:8]:
```python
for i in range(N-1, -1, -1):
```

File: python/src/code_70.py [hunk 6]
```diff
@@ -0,0 +1,13 @@
+class Solution:
+    def climbStairs(self, n: int) -> int:
+        if n <= 3:
```
Comment (dataprob) [inline python/src/code_70.py:3]: `if n <= 2: return n`

File: python/src/code_70.py [hunk 7]
```diff
@@ -0,0 +1,13 @@
...
+            return n
+
+        climbStairsSoln = [-1] * n
```
Comment (dataprob) [inline python/src/code_70.py:6]: I prefer `None` rather than `-1`. Here it's fine, but in another Q `-1` could be a valid value.

Comment (dataprob) [inline python/src/code_70.py:6]: `[None] * (n + 1)`, so the answer has nice index `n`

File: python/src/code_70.py [hunk 8]
```diff
@@ -0,0 +1,13 @@
...
+        climbStairsSoln[0] = 1
+        climbStairsSoln[1] = 2
+
+        for i in range(2, n):
```
Comment (dataprob) [inline python/src/code_70.py:10]: with 2 base cases, then it's `(3, n)`

File: python/src/code_66.py [hunk 9]
```diff
@@ -0,0 +1,20 @@
+from typing import List
```
Comment (dataprob) [inline python/src/code_66.py:1]: There's a space optimized dp solu, with 2 variables `prev, curr`, can try this as brain teaser. Your solu is good enough!

**Review by dataprob — COMMENTED** [review]
See my python comments. Will look at cpp tomorrow.

File: cpp/src/code_118.cpp [hunk 10]
```diff
@@ -0,0 +1,29 @@
+#include <vector>
+
+using namespace std;
+
+class Solution {
+public:
+    vector<vector<int>> generate(int numRows) {
+        if (numRows == 1) {
+            return {{1}};
+        } else if (numRows == 2) {
```
Comment (dataprob) [inline cpp/src/code_118.cpp:10]: Originally I did the same 2 cases too! But then I figured I only need `numRows == 1` case, can use for loop to handle `>= 2`.

Comment (dataprob) [reply cpp/src/code_118.cpp:10]: Better yet, you don't even need the `if`, just do `triangle = {{1}}` and change the for loop, so won't run for `numRows == 1`.

**Review by dataprob — COMMENTED** [review]
See my comments for cpp. My review is all done! Great work!

File: python/src/code_26.py
[the remark this answers is not in the document]
Comment (azalexzhang) [reply python/src/code_26.py:1]: I think `nums[:] = someValue` is the way to do it in-place (LeetCode accepted the solution with this assignment).

`nums[:] = list(set(nums))` fails with the test case `nums = [-1,0,0,0,0,3,3]` as they expect an array that is non-decreasing, and the order is lost (`nums` is set to `[0, 3, -1]`). The array would need to be sorted afterward.

File: python/src/code_58.py (hunk 2 above)
Comment (azalexzhang) [reply python/src/code_58.py:6]: On my end, `s.strip()` returns the same string but with all whitespace before and after removed. Therefore, if `s = "  fly me  to  the moon ", `strippedStr = "fly me  to  the moon" (no spaces at the beginning and end).

File: python/src/code_70.py (hunk 8 above)
Comment (azalexzhang) [reply python/src/code_70.py:10]: I guess you're suggesting a sort of 1-based indexing?

File: python/src/code_66.py (hunk 9 above)
Comment (dataprob) [reply python/src/code_66.py:1]: @azalexzhang Oh wait I meant to comment this in another file, somehow it's here and doesn't make sense, I'll try to see which file I originally wanted to comment.

Comment (dataprob) [reply python/src/code_66.py:1]: @azalexzhang It's Q70.

File: python/src/code_70.py (hunk 8 above)
Comment (dataprob) [reply python/src/code_70.py:10]: I have several solutions that passed, putting them here for clearity
**solu 1**
```py
class Solution:
    def climbStairs(self, n: int) -> int:
        if n <= 2:
            return n
        T = [None] * (n + 1)
        T[1] = 1
        T[2] = 2
        for i in range(3, n + 1):
            T[i] = T[i-1] + T[i-2]

        return T[n]
```
**solu 2**
```py
class Solution:
    def climbStairs(self, n: int) -> int:
        if n <= 1:
            return 1
        T = [None] * (n + 1)
        T[0] = T[1] = 1
        for i in range(2, n + 1):
            T[i] = T[i-1] + T[i-2]

        return T[n]
```
**solu 3**
```py
class Solution:
    def climbStairs(self, n: int) -> int:
        if n <= 1:
            return 1
        prev, curr = 1, 1
        for i in range(2, n + 1):
            temp = curr
            curr = prev + curr
            prev = temp

        return curr
```

File: cpp/src/code_118.cpp [hunk 11]
```diff
@@ -0,0 +1,29 @@
+#include <vector>
```
[the remark this answers is not in the document]
Comment (dataprob) [reply cpp/src/code_118.cpp:1]: To run `make`, see github codespaces guide in Git-ing.

**Review by dataprob — COMMENTED** [review]
Great work alex! Below are my follow-up comments.

**Comment by dataprob** [comment]
Also run lint locally, see [lint instructions](https://github.com/Prompt-ing/LeetCode-ing/blob/main/doc/onboarding/onboarding.md#how-to-format-the-code).

**Pull request merged**